### PR TITLE
build before test

### PR DESCRIPTION
**Pull request by MHBauer**
@duglin please check this for correctness of make file. How can I add a comment?

@rifung see if this helps run the tests faster.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -167,6 +167,7 @@ coverage: .init
 test: .init
 	@echo Running tests:
 	@for i in $(addprefix $(SC_PKG)/,$(TEST_DIRS)); do \
+	  $(DOCKER_CMD) go test -i $$i && \
```
Comment (duglin) [inline Makefile]: why did you remove the exit?

Comment (MHBauer) [inline Makefile]: This is a completely new line, scroll down one line

Comment (MHBauer) [inline Makefile]: What does the exit do? Do they both need an exit?

Comment (duglin) [inline Makefile]: ah missed that

Comment (MHBauer) [inline Makefile]: Is there an easier way to run both commands at one time? This was the easiest thing that worked.

Comment (duglin) [inline Makefile]: you could put them both in the same docker run cmd if you want

Comment (MHBauer) [inline Makefile]: How?

**Comment by MHBauer** [comment]
I'm actually iffy on this. It only helps if you do a `make test` as the first thing without doing a regular build.

File: Makefile (hunk 1 above)
Comment (duglin) [inline Makefile]: try:
```
$(DOCKER_CMD) go test -i $$i \&\& go test $$i || exit $$? ; \
```

**Comment by MHBauer** [comment]
For my runs it actually hurts if the build cache is up to date. Each final invocation of go test where it still needs to build an link before running takes 1.5 -2 seconds. So The final test time with a cache goes up 1.5s a test, adding ~20 sec to the `make test` time.

**Comment by rifung** [comment]
This seems to run faster for me but if it makes things slower for other things I'd rather we try to explore other solutions.

**Comment by MHBauer** [comment]
@rifung Have you done a `make build` first? Is it still faster after that?

I can see a case where someone is modifying their code and test at the same time, and it might be faster in that case as well if they don't do a full build.

**Comment by rifung** [comment]
I only tested it running make test multiple times in a row. It doesn't seem to go any faster after running a make build. (I should note I'm not doing any rigorous timing and just kind of looking at how fast they seem to run).

**Comment by MHBauer** [comment]
I think this is the way to go instead, unless people feel strongly.

**Comment by duglin** [comment]
@MHBauer what is this trying to fix?

I can never decide if "build" should be a dependency for "test" because there have been times when I wanted to run the tests w/o rebuilding because I was just messing around in some files.  But I'm ok if people want it.

**Comment by MHBauer** [comment]
If you run make test, without having done a build, it'll take X * Z time, where X is the number of test and Z is the time it takes to compile the k8s dependencies the package almost certainly imports. If you do a build first, you only have to compile the test changes. It become Z + (X*1).

**Comment by duglin** [comment]
freaky  LGTM

**Comment by krancour** [comment]
This doesn't make any sense. Building before testing does _not_ have any effect on time required to test.
